core/src/main/java/fr/sncf/osrd/api/InfraManager.java

```diff
@@ -138,14 +137,16 @@ private FullInfra downloadInfra(
             cacheEntry.transitionTo(InfraStatus.DOWNLOADING);
 
             RJSInfra rjsInfra;
+            String version;
             try (var response = httpClient.newCall(request).execute()) {
                 if (!response.isSuccessful())
                     throw new UnexpectedHttpResponse(response);
 
                 // Parse the response
                 logger.info("parsing the JSON of {}", request.url());
                 cacheEntry.transitionTo(InfraStatus.PARSING_JSON);
-
+                version = response.header("x-infra-version");
+                assert version != null : "missing x-infra-version header in railjson response";
                 rjsInfra = RJSInfra.adapter.fromJson(response.body().source());
```

Comment:
Why are we adding a custom HTTP header when we could simply use `rjsInfra.version` ?

Reply:
it's not the same: rjsInfra.version is the RailJSON format version, whereas the version in the header is the version of the infrastructure